pkg/proxy/iptables/proxier.go

```diff
@@ -332,12 +339,15 @@ func NewProxier(ipt utiliptables.Interface,
 		serviceChanges:           proxy.NewServiceChangeTracker(newServiceInfo, &isIPv6, recorder),
 		endpointsMap:             make(proxy.EndpointsMap),
 		endpointsChanges:         proxy.NewEndpointChangeTracker(hostname, newEndpointInfo, &isIPv6, recorder),
+		nodeMap:                  make(proxy.NodeMap),
+		nodeChanges:              proxy.NewNodeChangeTracker(),
```

Comment:
I think you're right, I'll make a change, only sync rules when there is a node label change.

Reply:
done

Reply:
I'm still struggling with the need for complexity here.  I admit it's been a while since I dug into it this deeply, but why do we need more than a `map[NodeName]map[string]string` and a mutex?

OnNodeAdd() { lock mutex; add node.labels to the map; unlock; call sync }

OnNodeDelete() { lock mutex; remove node from map }

OnNodeUpdate() { lock mutex; update node in map; unlock; call sync }

sync() will call FilterTopology...() which can take the mutex while evaluating.

No need for NodeChangeTracker, and all that related logic.  Maybe not even Node interface.  This seems a lot simpler - what doesn't work?

Reply:
Sounds reasonable, let me try it.

Reply:
> - receive events
-- ignore add (we don't have endpoints for it yet)
-- ignore delete (endpoints are going away or gone)

According to our discussion here, we will call sync only node labels was updated. And only update node labels cache when receive add and delete event.